src/Native/Runtime/RuntimeInstance.cpp

```diff
@@ -821,9 +821,10 @@ enum RuntimeHelperKind
 // The dictionary codegen expects a pointer that points at a memory location that points to the method pointer
 // Create indirections for all helpers used below
 
-#define DECLARE_INDIRECTION(HELPER_NAME) \
-    EXTERN_C void * HELPER_NAME; \
+#define DECLARE_INDIRECTION_TYPED(RET_TYPE, HELPER_NAME, ARGS) \
+    EXTERN_C RET_TYPE HELPER_NAME ARGS; \
     const PTR_VOID indirection_##HELPER_NAME = (PTR_VOID)&HELPER_NAME
+#define DECLARE_INDIRECTION(HELPER_NAME) DECLARE_INDIRECTION_TYPED(void, HELPER_NAME, ())
```

Comment:
Nit: It would not hurt to declare all these pointers with proper type. And then the macro name can stay as `DECLARE_INDIRECTION`

Reply:
I've made this change with the exception of `RhTypeCast_IsInstanceOf` and `RhTypeCast_CheckCast` as I think they want a `MethodTable *` which would be including `gcenv.base.h`, I can do that if that's better than the `void *` that these methods currently have?

Reply:
`RhTypeCast_IsInstanceOf` and `RhTypeCast_CheckCast` take `EEType *`

Reply:
Changed to `EEType *`